cmd/create/cluster/cmd.go

```diff
@@ -367,19 +361,6 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
-	// Creating non STS clusters is not supported
-	if creator.IsSTS && args.roleARN == "" {
-		reporter.Errorf("IAM STS credentials can only create STS clusters --role-arn cannot be blank")
-		reporter.Errorf("Your AWS credentials are returning an STS ARN: %s", creator.ARN)
-		os.Exit(1)
-	}
```

Comment:
This was added to prevent users from creating regular ROSA clusters with STS credentials I don't think we want to remove that.

Reply:
Agreed - since we cannot guarantee that the user will send the `--role-arn` in the initial payload, if the creator ARN is STS, then we make the Role ARN required

Reply:
The role ARN should be required though if the user is creating a STS cluster with non STS credentials.

We're preventing STS credentials from creating clusters that aren't STS. Non STS credentials should be able to create STS or non STS clusters.

Reply:
Correct, we are making the Role ARN required if the user uses STS credentials. However, whether the user is creating an STS cluster is determined by whether they enter a Role ARN or not, so it cannot be a required field.